CRM/Report/Form/Contact/Detail.php

```diff
@@ -975,4 +976,44 @@ public function alterComponentDisplay(&$componentRows) {
     }
   }
 
+  /**
+   * Function to alter from clause by adding checks for permissoned financial type.
+   *
+   * @param string $component
+   *
+   * @return string
+   */
+  public function addPermissionedFromClause($component) {
+    if (is_null($this->financialTypes)) {
+      return NULL;
+    }
+    $financialTypes = 0;
+    if (!empty($this->financialTypes)) {
+      $financialTypes = implode(',', array_keys($this->financialTypes));
+    }
+    $clause = "financial_type_id IN ({$financialTypes})";
+    $whereClause = NULL;
+
+    switch ($component) {
```

Comment:
@Edzelopez I think it would be better to go with something like 

if ($val == 'contribution') {
$whereClause = $this->getPermissionedWhereClauseForContribution() {

else {
$whereClause = ....

However, it's not a show-stopper.